src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer_ResourceStringsFormat.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Xml.Linq;
+using Analyzer.Utilities.PooledObjects;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers
+{
+    using PooledResourcesDataValueConcurrentDictionary = PooledConcurrentDictionary<string, ImmutableDictionary<string, (string value, Location location)>>;
+
+    public sealed partial class DiagnosticDescriptorCreationAnalyzer
+    {
+        private static bool HasResxAdditionalFiles(AnalyzerOptions options)
+        {
+            foreach (var file in options.AdditionalFiles)
+            {
+                if (string.Equals(".resx", Path.GetExtension(file.Path), StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static ImmutableDictionary<string, (string value, Location location)> GetOrCreateResourceMap(
+            AnalyzerOptions options,
+            string resourceFileName,
+            PooledResourcesDataValueConcurrentDictionary resourceMap,
+            CancellationToken cancellationToken)
+        {
+            Debug.Assert(HasResxAdditionalFiles(options));
+
+            resourceFileName += ".resx";
```

Comment:
Might be worth adding an assertion that the string doesn't already end in `.resx`, don't you think?